eox_nelp/pearson_vue/pipeline.py

```diff
@@ -0,0 +1,104 @@
+            - anonymous_user_id (str): An anonymized identifier for the user.
+            - username (str): The username of the user.
+            - first_name (str): The first name of the user.
+            - last_name (str): The last name of the user.
+            - email (str): The email address of the user.
+            - address (str): The mailing address of the user.
+            - city (str): The city of the user's address.
+            - country (str): The country code of the user's address.
+            - phone_number (int): The user's phone number in national format.
+            - phone_country_code (int): The country code of the user's phone number.
+            - mobile_number (int): The user's mobile number in national format (same as phone_number).
+            - mobile_country_code (int): The country code of the user's mobile number (same as phone_country_code).
+    """
+    user = User.objects.get(id=user_id)
+    profile = user.profile
+    phone = profile.phone_number
+    pn = phonenumbers.parse(phone) if phone.startswith("+") else phonenumbers.parse(phone, profile.country.code)
+    phone = str(pn.national_number)
+    phone_country_code = str(pn.country_code)
+
+    return {
+        "anonymous_user_id": anonymous_id_for_user(user, None),
+        "username": user.username,
+        "first_name": user.first_name,
+        "last_name": user.last_name,
+        "email": user.email,
+        "address": profile.mailing_address,
+        "city": profile.city,
+        "country": profile.country.code,
+        "phone_number": phone,
+        "phone_country_code": phone_country_code,
+        "mobile_number": phone,
+        "mobile_country_code": phone_country_code,
+    }
+
+
+def check_completion_metadata(user_id, course_id, **kwargs):
+    """Pipeline that check the case of completion cases on the pipeline execution. Also this pipe
+    has 3 behaviours depending the case:
+        - is_passing is true means the course is graded(passed) and dont needs this pipe validation.
+          The pipeline continues without changes.
+        - is_complete=True and is_graded=False pipeline should continue with uploaded metadata
+          (completed courses and not graded).
+        - Otherwise the pipeline will stop, for grading-courses the COURSE_GRADE_NOW_PASSED signal would act.
+
+    Args:
+        user_id (int): The ID of the user whose data is to be retrieved.
+        course_id (str): course_id to check completion or graded.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        dict: Pipeline dict
+    """
+    if kwargs.get("is_passing"):
+        return {}
+
+    is_complete, is_graded = get_completed_and_graded(user_id, course_id)
+
+    if is_complete and not is_graded:
+        return {
+            "is_complete": is_complete,
+            "is_graded": is_graded,
+        }
+
+    return {
+        "safely_pipeline_termination": True,
+    }
```

Comment:
```suggestion
    is_complete, is_graded = get_completed_and_graded(user_id, course_id)
    
    if kwargs.get("is_passing") or (is_complete and not is_graded):
        return {}

    return {
        "safely_pipeline_termination": True,
    }
```

Reply:
@andrey-canon I was thinking in this and I could skip the `get_completed_and_graded` call for grading course. That method uses modulestore,  so it skips a call in performance. 
Thinking more the better performance  approach is using  only the receiver of `COURSE_GRADE_NOW_PASSED` in order, to skip a lot of modulestore call thinking the completion of each unit, of each user, of each course =O xD. 
But we could still have both approaches

Reply:
But now this does't works  the None that you return is no compatible with the update method, so this will raise an exception when is_passing is True,  indeed now i don't like the suggestion since I looking for 
congruence in the  output  and you made me think that if we don't need anything when  is_passing is true  we don't need  is_complete and  is_graded at all, now this is my suggestion

```python
    safely_pipeline_termination = True
        
    if kwargs.get("is_passing"):
        safely_pipeline_termination = False 
    else:
        is_complete, is_graded = get_completed_and_graded(user_id, course_id)
        safely_pipeline_termination = not(is_complete and not is_graded)

    return {
        "safely_pipeline_termination": safely_pipeline_termination,
    }
```

Reply:
I think it could be compatible with the None response? the empty dict is overriding.
https://github.com/eduNEXT/eox-nelp/blob/e9f93d1bebddcdfe5e79530d289c16fa6c5fad7b/eox_nelp/pearson_vue/rti_backend.py#L41

WHat about:
```python3
    if kwargs.get("is_passing"):
        return 
    
    is_complete, is_graded = get_completed_and_graded(user_id, course_id)
    if is_complete and not is_graded:
        return

    return {
        "safely_pipeline_termination": True,
    }
```

Reply:
pylint error if I'm not wrong

Reply:
So with return None, you dont like it or empty dict. Some think is that the nextes pipe would have the safely_pipeline_termination kwarg in False defined

Reply:
Check [feacd71](https://github.com/eduNEXT/eox-nelp/pull/158/commits/feacd712aadc2e7cffdde9cbb938c76840b824e3)